xbmc/windowing/wayland/WinSystemWayland.cpp

```diff
@@ -783,7 +783,7 @@ void CWinSystemWayland::ProcessMessages()
   }
 }
 
-void CWinSystemWayland::ApplyShellSurfaceState(IShellSurface::StateBitset state)
+void CWinSystemWayland::ApplyShellSurfaceState(const IShellSurface::StateBitset& state)
```

Comment:
I don't think there is a point in passing `IShellSurface::StateBitset` by reference as `sizeof(IShellSurface::StateBitset) <= sizeof(size_t)`.

Reply:
Hmm, maybe cppcheck is wrong then?

```
xbmc/windowing/wayland/WinSystemWayland.cpp:786:75: performance: Function parameter 'state' should be passed by const reference. [passedByValue]
void CWinSystemWayland::ApplyShellSurfaceState(IShellSurface::StateBitset state)
                                                                          ^
xbmc/windowing/wayland/WinSystemWayland.cpp:965:113: performance: Function parameter 'state' should be passed by const reference. [passedByValue]
CWinSystemWayland::Sizes CWinSystemWayland::CalculateSizes(CSizeInt size, int scale, IShellSurface::StateBitset state, bool sizeIncludesDecoration)
                                                                                                                ^
xbmc/windowing/wayland/WinSystemWayland.cpp:1011:134: performance: Function parameter 'state' should be passed by const reference. [passedByValue]
CWinSystemWayland::SizeUpdateInformation CWinSystemWayland::UpdateSizeVariables(CSizeInt size, int scale, IShellSurface::StateBitset state, bool sizeIncludesDecoration)
```

Reply:
I've filed https://sourceforge.net/p/cppcheck/discussion/general/thread/d3299ff883/#9b5d for the potential cppcheck false positive.

Reply:
As the change has been accepted by Cppcheck please remove all the pass by reference so this can get merged.